crates/biome_js_analyze/src/analyzers/style/use_while.rs

```diff
@@ -65,51 +67,41 @@ impl Rule for UseWhile {
 
     fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
         let node = ctx.query();
-
-        // SAFETY: These tokens have been checked for errors in `run` already
-        let for_range = node.for_token().unwrap().text_trimmed_range();
-        let r_paren_range = node.r_paren_token().unwrap().text_trimmed_range();
+        let for_range = node.for_token().ok()?.text_trimmed_range();
+        let r_paren_range = node.r_paren_token().ok()?.text_trimmed_range();
 
         Some(RuleDiagnostic::new(
             rule_category!(),
             for_range.cover(r_paren_range),
             markup! {
-                "Use "<Emphasis>"while"</Emphasis>" loops instead of "<Emphasis>"for"</Emphasis>" loops."
+                "Use a "<Emphasis>"while"</Emphasis>" loop instead of a "<Emphasis>"for"</Emphasis>" loop."
             },
-        ))
+        ).note(markup! {
+            "A "<Emphasis>"while"</Emphasis>" loop is more readable than a "<Emphasis>"for"</Emphasis>" loop without initialization and update expressions."
```

Comment:
This is a stylistic rule, and the user needs to opt-in to use it. Given these statements, the "more readable" becomes implied. Plus, readable code is always personal and it boils down to an opinion. We should avoid these statements.

Maybe we should just say: "This rule prioritizes the use o while loops, over for lops without..."